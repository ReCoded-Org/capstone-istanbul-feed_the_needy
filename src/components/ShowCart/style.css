.couponCards {
  display: flex;
  align-items: center;
  text-align: center;
}

.ant-btn.ant-btn-primary.cardButton {
  display: inline-block;
  align-items: center;
  text-align: center;
  font-size: xx-large;
  height: 1.8em;
  width: 1.5em;
  border-radius: 0.3em;
  background-color: #627280;
}

.ant-btn.ant-btn-primary.cardDeleteButton {
  font-size: small;
  font-weight: bold;
  border: none;
  width: auto;
  height: auto;
  border-radius: 0.3em;
  margin-bottom: 1.4em;
  color: white;
  background-color: rgb(100, 6, 6);
}

.couponAmount {
  display: inline-block;
  text-align: center;
  font-size: xx-large;
  height: 1.5em;
  width: 2em;
  color: white;
  background-color: #ffaa05;
}

.totalAmount {
  font-size: xx-large;
  font-weight: bold;
  margin-top: 1.2em;
}

.clearCartButton {
  width: auto;
  height: auto;
  font-size: large;
  background-color: #627280;
  color: white;
}

.checkoutButton {
  width: auto;
  height: auto;
  font-size: large;
  color: white;
  background-color: #e17007;
}
