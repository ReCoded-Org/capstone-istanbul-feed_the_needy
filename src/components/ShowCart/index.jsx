import React, { useState, useEffect } from "react";
import "./style.css";
import { Button, Row, Col, Empty } from "antd";
import { useTranslation } from "react-i18next";
import SingleCoupon from "../SingleCoupon";
import CheckoutDrawer from "../CheckoutDrawer";

const ShowCart = ({ isTesting }) => {
  const { t } = useTranslation();
  const [cart, setCart] = useState([]);
  const [totalAmount, setTotalAmount] = useState(0);
  const [drawerVisible, setDrawerVisible] = useState(false);

  const showDrawer = () => {
    setDrawerVisible(true);
  };

  const adjustCouponQuantity = (id, operation) => {
    const modifiedCart = cart.map((coupon) => {
      if (coupon.id === id) {
        operation === "increment"
          ? (coupon.quantity += 1)
          : (coupon.quantity -= 1);
        return coupon;
      }
      return coupon;
    });
    setCart(modifiedCart);
    const modifiedCartJson = JSON.stringify(modifiedCart);
    localStorage.setItem("cart", modifiedCartJson);
  };

  useEffect(() => {
    const cartJSON = localStorage.getItem("cart");
    const cartParsed = JSON.parse(cartJSON);
    if (cartParsed) {
      setCart(cartParsed);
      setTotalAmount((totalAmount) => {
        let sum = 0;
        for (let i = 0; i < cartParsed.length; i++) {
          sum += cartParsed[i].amount * cartParsed[i].quantity;
        }
        return sum;
      });
    }
  }, []);

  const changeAmount = (amount, operation) => {
    if (operation === "increment") {
      setTotalAmount(totalAmount + amount);
    } else if (operation === "decrement") {
      setTotalAmount(totalAmount - amount);
    }
  };

  const deleteSingleCoupon = (id) => {
    setCart((prevCart) => {
      let newCart = [...prevCart];
      newCart = newCart.filter((coupon) => coupon.id !== id);
      localStorage.setItem("cart", JSON.stringify(newCart));
      return newCart;
    });
  };

  return (
    <>
      <br />
      <Row gutter={[24, 40]} justify="space-around">
        {cart.length < 1 ? (
          <Empty
            image={Empty.PRESENTED_IMAGE_SIMPLE}
            imageStyle={{
              height: 100,
            }}
            description={<span>Cart is empty</span>}
          />
        ) : (
          cart.map((coupon) => (
            <SingleCoupon
              coupon={coupon}
              changeAmount={changeAmount}
              deleteSingleCoupon={deleteSingleCoupon}
              setTotalAmount={setTotalAmount}
              adjustCouponQuantity={adjustCouponQuantity}
            />
          ))
        )}
      </Row>
      <Row gutter={[48, 48]} justify="center">
        <Col className="totalAmount">
          {t("cartCont.showCartComp.totalAmount")}
          {totalAmount}
          <span>â‚º</span>
        </Col>
      </Row>
      <br />
      <br />
      <Row gutter={[48, 48]} justify="center" align="bottom">
        <Col>
          {isTesting ? (
            <button type="button">{}</button>
          ) : (
            <Button
              onClick={() => {
                localStorage.removeItem("cart");
                setCart([]);
                setTotalAmount(0);
              }}
              className="clearCartButton"
            >
              {t("cartCont.showCartComp.clearCart")}
            </Button>
          )}
        </Col>
        <Col>
          {isTesting ? (
            <button type="button">{}</button>
          ) : (
            <Button
              onClick={showDrawer}
              className="checkoutButton"
              type="primary"
            >
              {t("cartCont.showCartComp.goToCheckout")}
            </Button>
          )}
        </Col>
      </Row>
      <CheckoutDrawer
        cart={cart}
        setDrawerVisible={setDrawerVisible}
        drawerVisible={drawerVisible}
      />
    </>
  );
};

export default ShowCart;
