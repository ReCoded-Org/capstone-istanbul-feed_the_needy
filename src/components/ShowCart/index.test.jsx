import React from "react";
import renderer from "react-test-renderer";
import ShowCart from "./index";

// Fix for TypeError: window.matchMedia is not a function
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

it("ShowCart renders correctly", () => {
  const tree = renderer.create(<ShowCart isTesting />).toJSON();
  expect(tree).toMatchSnapshot();
});
