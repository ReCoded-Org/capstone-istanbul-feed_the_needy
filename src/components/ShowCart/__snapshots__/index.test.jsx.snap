// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowCart renders correctly 1`] = `
Array [
  <br />,
  <div
    className="ant-row ant-row-space-around"
    style={
      Object {
        "marginBottom": 20,
        "marginLeft": -12,
        "marginRight": -12,
        "marginTop": -20,
      }
    }
  />,
  <div
    className="ant-row ant-row-center"
    style={
      Object {
        "marginBottom": 24,
        "marginLeft": -24,
        "marginRight": -24,
        "marginTop": -24,
      }
    }
  >
    <div
      className="ant-col totalAmount"
      style={
        Object {
          "paddingBottom": 24,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 24,
        }
      }
    >
      cartCont.showCartComp.totalAmount
      0
      <span>
        â‚º
      </span>
    </div>
  </div>,
  <br />,
  <br />,
  <div
    className="ant-row ant-row-center ant-row-bottom"
    style={
      Object {
        "marginBottom": 24,
        "marginLeft": -24,
        "marginRight": -24,
        "marginTop": -24,
      }
    }
  >
    <div
      className="ant-col"
      style={
        Object {
          "paddingBottom": 24,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 24,
        }
      }
    >
      <button
        type="button"
      />
    </div>
    <div
      className="ant-col"
      style={
        Object {
          "paddingBottom": 24,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 24,
        }
      }
    >
      <button
        type="button"
      />
    </div>
  </div>,
]
`;
